sdk/core/Azure.Core.TestFramework/src/Instrumentation/DiagnosticScopeValidatingInterceptor.cs

```diff
@@ -260,6 +261,58 @@ internal static async ValueTask<T> ValidateDiagnosticScope<T>(Func<ValueTask<(T
             return result;
         }
 
+        private static void CheckAttributes(Activity activity, bool strict)
+        {
+            foreach (var tag in activity.TagObjects)
+            {
+                if (tag.Key == "kind" || tag.Key.StartsWith("otel.") || tag.Key == "requestId" || tag.Key == "serviceRequestId")
+                {
+                    // TODO: these are populated on DiagnosticSource path. We should rewrite this to ActivitySource and remove those.
```

Comment:
Curious: when will we address this?  Should we have an issue tracking it?

Reply:
great point, added https://github.com/Azure/azure-sdk-for-net/issues/39782